src/utils/contract.tsx

```diff
@@ -64,3 +64,41 @@ export const getContractByAgreementId = (
     ),
   )
 }
+
+export const getFirstMasterInception = (
+  contracts: ReadonlyArray<Contract>,
+): Date | null => {
+  const masterInceptions = contracts
+    .filter((contract) => !!contract.masterInception)
+    .map((contract) => contract.masterInception)
+  if (masterInceptions.length === 0) {
+    return null
+  }
+  return masterInceptions.reduce((a, b) => {
+    if (a < b) {
+      return a
+    }
+    return b
```

Comment:
I know I wrote this one but 😄 
```suggestion
    return a < b ? a : b
```